resources/js/common/utils/l10n/formatNumber.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Number.prototype.toLocaleString() can be quite slow if called frequently.
+ * User Intl.NumberFormat with a given locale is generally much faster.
+ *
+ * TODO pass in the user's locale from the `auth` object in page props
+ *
+ * @returns A localized number. eg: 12345 -> "12,345"
+ */
```

Comment:
Similarly, this adds number separators to numbers in a localized fashion. Currently, this defaults to en-US, but should ultimately be driven by a user setting.